app/resources/v1/qc_reception_resource.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module V1
+  # QcReceptionResource
+
+  # Steps:
+  # 1. Create QcReception
+  # 2. Create QcResults
+  # 3. Publish qc result messages
+  class QcReceptionResource < JSONAPI::Resource
+    attributes :qc_results_list, :source
+
+    # The after_create happens in a transaction, and since there are
+    # multiple callbacks, they are executed is reverse order.
+    # create_qc_results! is executed first, followed by publish_messages
```

Comment:
Thanks for the comment :)

Reply:
Thinking about this... I think it would be good to make sure the order doesn't change in future because of new releases of rails. Could we move these under a single method that will call the 2 callbacks in the order we want?